api.go

```diff
@@ -830,3 +834,105 @@ func (c Client) makeTargetURL(bucketName, objectName, bucketLocation string, que
 
 	return u, nil
 }
+
+// isVirtualHostedStyle - returns true if server supports
+// virtual hosted-style urls
+func (c *Client) isVirtualHostedStyle(bucketName string) bool {
+	if c.endpointURL == sentinelURL {
+		return false
+	}
+	// bucketName can be valid but '.' in the hostname will fail SSL
+	// certificate validation. So do not use host-style for such buckets.
+	if c.endpointURL.Scheme == "https" && strings.Contains(bucketName, ".") {
+		return false
+	}
+	return c.isVirtualHostStyle
+}
+
+// setVirtualHostStyle - auto probes if server supports virtual
+// hosted-style urls
+func (c *Client) setVirtualHostStyle() {
```

Comment:
Can you add more comments about the logic of this function ?

Reply:
updated function with comments